### DISCUSSION TITLE
What constitutes a cache key for fetch cache?

### DISCUSSION THREAD

**Discussion by tomdohnal** [vercel/next.js#61498] [App Router]
Is it some combination of url, HTTP method and headers? Or something else? 
I couldn't find the answer anywhere. Thank you! 🙏

**Comment by icyJoseph** [comment]
Hi,

You can look at it here, https://github.com/vercel/next.js/blob/e02397d26473ac41a3d6dd6a396769b584dd0e49/packages/next/src/server/lib/patch-fetch.ts#L296, but it is kind of an implementation detail, why do you need to know?

## Edit

I had pasted the wrong link 😅 

https://github.com/vercel/next.js/blob/canary/packages/next/src/server/lib/incremental-cache/index.ts#L381

Anyhow...

**Reply by gpxl-dev** [reply]
I landed in this discussion thread because I was trying to understand if it was customisable... 

I'm building an app that reads data from a blockchain using JSON RPC. The RPC requests are all `POST` requests, and each one has a sequential `id` in the body as well as a method name and some data.

This makes it pretty hard for next to cache anything because each request has a different ID, but actually in most (all?) cases it's perfectly valid to cache a request with a different ID but with everything else the same. Ideally I was looking for a way to pass a function that would generate the cache key or some predicate...

**Reply by icyJoseph** [reply]
Right, I guess that's where the React cache would come in? And you'd cache by parameters not by request information. 

- https://react.dev/reference/react/cache#preload-data

**Reply by gpxl-dev** [reply]
react cache is per request (to the server) only though.

Ideally I'd like to persist data used to render the page for Alice when rending it for Bob.
